server/src/main/java/org/elasticsearch/common/util/CuckooFilter.java

```diff
@@ -142,18 +153,26 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeVInt(entriesPerBucket);
         out.writeVInt(count);
         out.writeVInt(evictedFingerprint);
-
-        data.save(new DataOutput() {
-            @Override
-            public void writeByte(byte b) throws IOException {
-                out.writeByte(b);
-            }
-
-            @Override
-            public void writeBytes(byte[] b, int offset, int length) throws IOException {
-                out.writeBytes(b, offset, length);
+        if (out.getVersion().before(Version.V_8_0_0)) {
+            // This is probably slow but it should only happen if we have a mixed clusters (e.g during upgrade).
+            PackedInts.Mutable mutable = PackedInts.getMutable(numBuckets * entriesPerBucket, bitsPerEntry, PackedInts.COMPACT);
+            for (int i = 0; i < count; i++) {
+                mutable.set(i, data.get(i));
             }
-        });
+            mutable.save(new DataOutput() {
```

Comment:
Similarly, `save()` is no longer in the lucene 9 API

Reply:
This will be removed once we backport the change to 7.x